src/libsyntax/ext/build.rs

```diff
@@ -429,7 +429,7 @@ impl<'a> AstBuilder for ExtCtxt<'a> {
         self.ty_path(
             self.path_all(DUMMY_SP,
                           true,
-                          self.std_path(&["option", "Option"]),
+                          self.std_path(&[sym::option, sym::Option]),
```

Comment:
cc @Centril this file is another candidate for lang item paths

Reply:
Aaaw... I was hoping hard coded paths would be limited to HIR =P.

Reply:
This function is actually unused.

Reply:
The `AstBuilder` is generally used for (legacy) proc macros and they face the same issues as modern proc macros - "how to find `serde::foo`", except it's the standard library and not `serde` in this case.

Ultimately, we should rather resolve this question for all macros, than hard-code more lang items for standard library macros specifically.

Reply:
@petrochenkov I may have spoken too soon. ^^ I think these being ~macros is a good argument for not using lang items. Ideally, we would migrate these things to actual proc macros if possible.

Reply:
I will remove the unused function. Other than that, I will leave this unchanged, because it's not clear you two have agreement on an alternative.

Reply:
You mean us three? I agree with petrochenkov on this one after further consideration -- lowering and macro expansion are different things. :)